main.go

```diff
@@ -81,29 +83,33 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = (&controllers.WorkflowReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("Workflow"),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "Workflow")
-		os.Exit(1)
-	}
-
-	if os.Getenv("ENVIRONMENT") == "kind" {
-		if err = (&controllers.ClientMountReconciler{
+	if mode == "controller" {
```

Comment:
I'd add a default handler for any unsupported mode like what nnf-sos does https://github.com/NearNodeFlash/nnf-sos/blob/0b66626501205edee4565444a30987976d15931c/main.go#L103